### PR TITLE
jwhois: unbreak darwin

### PR DESCRIPTION
**Pull request by ofalvai**
## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [X] x86_64-darwin
  - [X] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/tools/networking/jwhois/default.nix [hunk 1]
```diff
@@ -38,6 +38,8 @@ stdenv.mkDerivation (finalAttrs: {
   # TODO: this should probably be fixed at a lower level than this?
   env.NIX_CFLAGS_COMPILE = lib.optionalString stdenv.hostPlatform.isDarwin "-Wno-undef-prefix";
```
Comment (emilazy) [inline pkgs/tools/networking/jwhois/default.nix:39]: Is this still necessary?

**Review by emilazy — APPROVED** [review]
Thanks!

File: pkgs/tools/networking/jwhois/default.nix (hunk 1 above)
Comment (ofalvai) [reply pkgs/tools/networking/jwhois/default.nix:39]: Yes, I tried removing it but it's load bearing 🫠 
This is the full error, but I can't make sense of it:
```
jwhois> make[3]: Entering directory '/private/tmp/nix-build-jwhois-4.0.drv-0/jwhois-4.0/gl'
jwhois> clang -DHAVE_CONFIG_H -I. -I. -I..     -g -O2 -c dummy.c
jwhois> clang -DHAVE_CONFIG_H -I. -I. -I..     -g -O2 -c asnprintf.c
jwhois> clang -DHAVE_CONFIG_H -I. -I. -I..     -g -O2 -c printf-args.c
jwhois> clang -DHAVE_CONFIG_H -I. -I. -I..     -g -O2 -c printf-parse.c
jwhois> clang -DHAVE_CONFIG_H -I. -I. -I..     -g -O2 -c vasnprintf.c
jwhois> In file included from vasnprintf.c:38:
jwhois> In file included from ./stdio.h:30:
jwhois> ///nix/store/4s8z8il6zyq77ixy4b8kfzwsnz90vsrm-apple-sdk-11.3/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/stdio.h:220:5: error: 'TARGET_OS_IPHONE' is not defined, evaluates to 0 [-Werror,-Wundef-prefix=TARGET_OS_]
jwhois> #if TARGET_OS_IPHONE
```

Comment (emilazy) [reply pkgs/tools/networking/jwhois/default.nix:39]: Yeah… I looked into this a while ago and couldn’t quite figure out why that stuff isn’t reliably being defined. I forget my exact conclusions though. There’s a bunch of this throughout the tree so it’s not blocking here, just would be nice to get it properly fixed at some point.

**Pull request merged**